x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsAction.java

```diff
@@ -98,7 +98,11 @@ protected void masterOperation(
             Settings.Builder decommissionAlertSetting = Settings.builder().put(Monitoring.MIGRATION_DECOMMISSION_ALERTS.getKey(), true);
             client.admin()
                 .cluster()
-                .prepareUpdateSettings()
+                .prepareUpdateSettings(
+                    request.masterNodeTimeout(),
+                    /* TODO expose separate ack timeout? use masterNodeTimeout() for now */
```

Comment:
Do you want to start a meta issue with links to all of these potential follow-up items? I get the sense this PR isn't going to *solve* these, but it does identify them.

Reply:
The point of https://github.com/elastic/elasticsearch/issues/107984 is to make these choices more explicit in the calling code, and hopefully to reduce the rate at which we add new things like this, but I expect a meta-issue to track any followups would just sit unaddressed for years anyway. It's just not very important. We've had these fixed (or at least confused) timeouts for a long time.